### ISSUE TITLE
User Switch stops process

### ISSUE THREAD

**Issue by geephile** [Spotifyd/spotifyd#902]
As soon as a 2nd user tries to connect the spotifyd process stops without an error (looks like a clean shutdown). 
Not sure if this is a bug or a "feature" or an "works as intended" 

**Environment**
- OS: Debian 11 (Bullseye/Testing) with pulseaudio backend
- Spotifyd: 0.30 (full and slim prebuild)
- Mutiple users with their own spotify premium accounts 

**Expected behaviour:**
- Run spotifyd without -u/-p options
- User A connects
- User A issues command to start music 
- Spotifyd starts playing music from User A
- User B Connects
- User B issues command to start music
- Spotifyd stops playing music from User A
- Spotifyd starts playing music from User B

**Observed behaviour**
_Situation 1_
- Running spotifyd without -u/-p options
- First user (A or B) that connects can play music
- as soon as the other user connects after that, the spotifyd process closes

_Situation 2_
- Running spotifyd with -u/-p options (credentails of user A)
- as soon as user (B) connects the spotifyd process closes
- as long as a user with the -u/-p credentials connects he (or she) can play music

**Logs (last bit)**
<details><summary>Click to show logs</summary>
```
drop Spirc[0]
Shutting down player thread ...
drop Player[0]
drop Session[0]
drop AudioKeyManager
drop ChannelManager
drop MercuryManager
shutdown; state=pool::State { lifecycle: Running, num_futures: 0 }
  -> transitioned to shutdown
  -> shutting down workers
dropping I/O source: 3
deregistering handle with poller
dropping I/O source: 1
deregistering handle with poller
dropping I/O source: 2
deregistering handle with poller
dropping I/O source: 0
drop Dispatch
deregistering handle with poller
dropping I/O source: 5
```
</details>

**Comment by robinvd** [comment]
I have found this as well, but no quick fix for this yet

**Comment by geephile** [comment]
Version 0.3.2 has the same issue

**Comment by geephile** [comment]
https://github.com/librespot-org/librespot/issues/579 
The original library seems to have the same issue

**Comment by eladyn** [comment]
If I'm not mistaken, this has been fixed in #1059.

**Comment by geephile** [comment]
I can confirm that this issue is fixed (at least in a compile from the current source)
